cookies/resources/helpers.py

```diff
@@ -1,46 +1,48 @@
-import urlparse
+from six.moves.urllib.parse import parse_qs
+
+from wptserve.utils import isomorphic_encode
 
 def setNoCacheAndCORSHeaders(request, response):
     """Set Cache-Control, CORS and Content-Type headers appropriate for the cookie tests."""
-    headers = [("Content-Type", "application/json"),
-               ("Access-Control-Allow-Credentials", "true")]
+    headers = [(b"Content-Type", b"application/json"),
+               (b"Access-Control-Allow-Credentials", b"true")]
 
-    origin = "*"
-    if "origin" in request.headers:
-        origin = request.headers["origin"]
+    origin = b"*"
+    if b"origin" in request.headers:
+        origin = request.headers[b"origin"]
 
-    headers.append(("Access-Control-Allow-Origin", origin))
+    headers.append((b"Access-Control-Allow-Origin", origin))
     #headers.append(("Access-Control-Allow-Credentials", "true"))
-    headers.append(("Cache-Control", "no-cache"))
-    headers.append(("Expires", "Fri, 01 Jan 1990 00:00:00 GMT"))
+    headers.append((b"Cache-Control", b"no-cache"))
+    headers.append((b"Expires", b"Fri, 01 Jan 1990 00:00:00 GMT"))
 
     return headers
 
 def makeCookieHeader(name, value, otherAttrs):
     """Make a Set-Cookie header for a cookie with the name, value and attributes provided."""
     def makeAV(a, v):
-        if None == v or "" == v:
+        if None == v or u"" == v:
             return a
         return "%s=%s" % (a, v)
 
     # ensure cookie name is always first
-    attrs = ["%s=%s" % (name, value)]
-    attrs.extend(makeAV(a, v) for (a,v) in otherAttrs.iteritems())
-    return ("Set-Cookie", "; ".join(attrs))
+    attrs = [u"%s=%s" % (name, value)]
+    attrs.extend(makeAV(a, v) for (a,v) in otherAttrs.items())
+    return (b"Set-Cookie", isomorphic_encode(u"; ".join((attrs))))
 
 def makeDropCookie(name, secure):
-    attrs = {"MaxAge": 0, "path": "/"}
+    attrs = {u"MaxAge": 0, u"path": u"/"}
     if secure:
-        attrs["secure"] = ""
-    return makeCookieHeader(name, "", attrs)
+        attrs[u"secure"] = u""
+    return makeCookieHeader(name, u"", attrs)
 
 def readParameter(request, paramName, requireValue):
```

Comment:
Below we have `readCookies` that always returns bytes (because of `request.cookies`). I think we should call `isomorphic_encode` here to make `readParameter` binary-only as well for consistency (with both `readCookies` and `request.GET`).

Reply:
Code updated